.github/workflows/ruby.yml

```diff
@@ -1,30 +1,31 @@
 name: Puma
 
-on:
-  push:
-    branches:
-      - '*'
-  pull_request:
-    branches:
-      - '*'
+on: [push, pull_request]
 
 jobs:
   build:
     name: >-
-      OS: ${{ matrix.os }}  Ruby: ${{ matrix.ruby }}
+      ${{ matrix.os }}  Ruby: ${{ matrix.ruby }}
+    env:
+      CI: true
+      TESTOPTS: -v
+
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ 'ubuntu-18.04', 'macos', 'windows-latest' ]
-        ruby: [ '2.4.x', '2.5.x', '2.6.x' ]
+        os: [ ubuntu-18.04, macos, windows-latest ]
+        ruby: [ 2.3, 2.4, 2.5, 2.6, 2.7 ]
+        exclude:
+          - os: windows-latest
+            ruby: 2.3
+
     steps:
     - name: repo checkout
-      uses: actions/checkout@v1
-      with:
-        fetch-depth: 10
+      uses: actions/checkout@v2
+
     - name: load ruby
-      uses: actions/setup-ruby@v1
+      uses: eregon/use-ruby-action@master
```

Comment:
I'll contribute some background links on this change for anyone curious to know more

* The repo for this is action is https://github.com/eregon/use-ruby-action
* It is using https://github.com/eregon/ruby-install-builder
* And it might become next version of the official action, see more in https://github.com/actions/setup-ruby/issues/44

Reply:
@MSP-Greg Could you update to `uses: eregon/use-ruby-action@v1` here?

Reply:
```suggestion
      uses: eregon/use-ruby-action@v1
```

Reply:
Done